sky/optimizer.py

```diff
@@ -806,6 +823,24 @@ def _cloud_in_list(cloud: clouds.Cloud, lst: List[clouds.Cloud]) -> bool:
     return any(cloud.is_same_cloud(c) for c in lst)
 
 
+def _generate_launchables_with_region_zones(
+        resources: resources_lib.Resources) -> List[resources_lib.Resources]:
+    assert resources.is_launchable()
+    launchables = []
+    for region, zones in resources.get_valid_region_zones():
+        if (resources.region is not None and region.name != resources.region):
+            continue
+        if not resources.use_spot:
```

Comment:
Should add your comment https://github.com/skypilot-org/skypilot/pull/975/files#r943893318 here.

Btw, if we have batching implemented, then we do not need to special-case here, right? Basically on-demand requests can also have a list of `(r, z1), (r, z2), ...` and then batching will detect that they all have the same prices. Worth adding a comment?

Reply:
Added. PTAL.